File: twitter-nwt/Models/Tweet.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace twitter_nwt.Models
{
    public class Tweet
    {
        [Key]
        public int Id { get; set; }

        [StringLength(140)]
        public string Text { get; set; }

        [DataType(DataType.DateTime)]
        public DateTime TimeWhenTweeted { get; set; }

        [JsonIgnore]
        public User User { get; set; }

        [InverseProperty("FavoritedTweets")]
        [JsonIgnore]
        public ICollection<User> UsersThatFavorited { get; set; }

        [InverseProperty("Tweets")]
        [JsonIgnore]
        public ICollection<Hashtag> Hashtags { get; set; }

        [JsonIgnore]
        public ICollection<Comment> Comments { get; set; }
    }
}

File: twitter-nwt/Assets/app/Common/Models/Tweet.ts
import {User} from "./User"
import {Hashtag} from "./Hashtag"
import {Comment} from "./Comment"

export class Tweet {
    public id: number;
    public user: User;
    public timeWhenTweeted: Date;
    public text: string;
    public hashtags: Hashtag[];
    public comments: Comment[];

    constructor(id: number = Math.random(), user: User, timeWhenTweeted: Date, text: string, hashtags: Hashtag[] = [], comments: Comment[] = []) {
        this.id = id;
        this.user = user;
        this.timeWhenTweeted = timeWhenTweeted;
        this.text = text;
        this.hashtags = hashtags;
        this.comments = comments;
    }
}